Mathlib/MeasureTheory/Constructions/Pi.lean

```diff
@@ -753,6 +753,34 @@ theorem volume_measurePreserving_piCongrLeft (α : ι → Type*) (f : ι' ≃ ι
     MeasurePreserving (MeasurableEquiv.piCongrLeft α f) volume volume :=
   measurePreserving_piCongrLeft (fun _ ↦ volume) f
 
+theorem measurePreserving_arrowProdEquivProdArrow (α β γ : Type*) [MeasurableSpace α]
+    [MeasurableSpace β] [Fintype γ] (μ : γ → Measure α) (ν : γ → Measure β) [∀ i, SigmaFinite (μ i)]
+    [∀ i, SigmaFinite (ν i)] :
+    MeasurePreserving (MeasurableEquiv.arrowProdEquivProdArrow α β γ)
+      (.pi fun i ↦ (μ i).prod (ν i))
+        ((Measure.pi fun i ↦ μ i).prod (Measure.pi fun i ↦ ν i)) where
+  measurable := (MeasurableEquiv.arrowProdEquivProdArrow α β γ).measurable
+  map_eq := by
+    refine (FiniteSpanningSetsIn.ext ?_ (isPiSystem_pi.prod isPiSystem_pi)
+      ((FiniteSpanningSetsIn.pi fun i ↦ (μ i).toFiniteSpanningSetsIn).prod
+      (FiniteSpanningSetsIn.pi (fun i ↦ (ν i).toFiniteSpanningSetsIn))) ?_).symm
+    · refine (generateFrom_eq_prod generateFrom_pi generateFrom_pi ?_ ?_).symm
+      exact (FiniteSpanningSetsIn.pi (fun i ↦ (μ i).toFiniteSpanningSetsIn)).isCountablySpanning
+      exact (FiniteSpanningSetsIn.pi (fun i ↦ (ν i).toFiniteSpanningSetsIn)).isCountablySpanning
```

Comment:
I hope the new linter will flag this as well, but it looks like you forgot some `·`s